sync/src/block/extension.rs

```diff
@@ -440,8 +440,18 @@ impl Extension {
     }
 
     fn create_headers_response(&self, start_number: BlockNumber, max_count: u64) -> ResponseMessage {
+        let highest_header = self.client.highest_header();
         let headers = (0..max_count)
-            .map(|number| self.client.block(&BlockId::Number(start_number + number)))
+            .map(|number| {
+                let height = start_number + number;
+                let block_id = if highest_header.number() == height {
+                    // We could not get highest block with height
+                    BlockId::Hash(highest_header.hash())
```

Comment:
How about add a comment about relation between `highest_header_number` and `best_block_number` in here?
Currently, every consensus except Tendermint ensures `best_block_num = highest_block_num` and Tendermint ensures `best_block_num + 1 >= highest_block_num`.

Reply:
Good

Reply:
I added a comment.